src/sources.cpp

```diff
@@ -270,7 +270,13 @@ static void src_vol_chunkloop(fields_chunk *fc, int ichunk, component c, ivec is
     loc += shift * (0.5 * inva);
 
     vec rel_loc = loc - data->center;
-    amps_array[idx_vol] = IVEC_LOOP_WEIGHT(s0, s1, e0, e1, 1) * amp * data->A(rel_loc);
+
+    // in cylindrical coordinates, we need to account for the r term in the integral.
+    if (fc->gv.dim == Dcyl)
+      amps_array[idx_vol] =
+          IVEC_LOOP_WEIGHT(s0, s1, e0, e1, dV0 + dV1 * loop_i2) * amp * data->A(rel_loc);
+    else
+      amps_array[idx_vol] = IVEC_LOOP_WEIGHT(s0, s1, e0, e1, 1) * amp * data->A(rel_loc);
```

Comment:
This can block can be more cleanly expressed as:
```cpp
double interp_weight = fc->gv.dim == Dcyl ? dV0 + dV1 * loop_i2 : 1.0;
amps_array[idx_vol] = IVEC_LOOP_WEIGHT(s0, s1, e0, e1, interp_weight) * amp * data->A(rel_loc);
```

Reply:
Sure, but let's make sure things work before we golf the code.